### ISSUE TITLE
What do predit_fake_summary and predit_real_summary show?

### ISSUE THREAD

**Issue by Cristy94** [affinelayer/pix2pix-tensorflow#32]
So there are the `predit_fake_summary` and `predict_real_summary` in the Tensorboard Images tab (next to the input/output/target images). 
From my understanding this should show what parts of the input image are considered "real" by the discriminator and what pars of the fake (generated) image are considered "real".

The problem is that most of the times `predict_fake` is an entirely black image while `predict_real` is a blank white image. What should I learn from this?

Also, is there any way to output the generated image by the cGAN generator to tensorboard? Or is that output the actual final output of the network?

**Comment by Cristy94** [comment]
So, after some reading, I think I am able to answer my own questions:

- `predict_fake_summary` shows for each pixel whether or not the discriminator thinks it's real or fake. If the image is all black it means that the discriminator confirmed the image to be a fake, if it is white it means that the fake image fooled the discriminator.
- `predict_fake_summary` does the same thing but for the real output image. If this image is all white it means that the discriminator successfully confirmed that the real image is real.
Now, if the images are not completely white or black it means that it considers some areas of that image to be real or fake, where white means real and black means fake.
- Yes, the **output** generated by the cGAN generator is the actual output of our network. For the given input the generator tries to "fool" the discrminator and that same image that was sent to the discriminator is sent as the final output of the network.

Correct me if I am wrong.

**Comment by tonmoyborah** [comment]
Hi @Cristy94 
Nice explanation. I was struggling with this too. 
I have another doubt. Are the outputs shown in tensorboard images tab the outputs of the generator? If so, is there anyway the generator knows/sees in some way anything about the target?
I am trying to train a sketch to image kind of dataset where I give sketches (say, of handbags) as inputs and train the network to output real looking images of handbags. But in one of my experiments I provided 500 images of brown handbags and 500 images of blue handbags shuffled together. I only provide sketches as input. But the output and target images shown in tensorboard always gets the correct color. I mean, for blue handbag, the output is always a blurred blue handbag even in the initial epochs. From the sketches alone, it is impossible to say which should be blue and which is brown. How do you think  the network is getting this?
Thanks!

**Comment by julien2512** [comment]
@Cristy94 
> LE: Those two images are actually 30x30 pixels, so they do not show the difference between each pixel, but between each "patch" (an submatrix of pixels)

Not really. They are the result of multiple conv layer over inputs/targets and input/output that is said a discriminator.
You must know conv layers are best on finding characteristics on images.

The characteristic we want to find here is whether the image is a fake or not, using 2 type of data : 
1. inputs vs targets => we know the result need to be [1,1,1,1,1,...] (30x30)
2.  inputs vs outputs => we know the result need to be [0,0,0,0,0,...] (30x30)

@tonmoyborah 
Try these mind steps :
1. Try a deep learning on 2 images only, and see the results.
2. Think about the size of your neural network (250Mo or something) and the total size of your images.
3. The generator do not use the targets to produce the output. But the process of learning is using it.
4. With a too small buch of data, the neural network can learn correlations that won't exist with a larger bunch.

Regards,
